internal/core/cppapi_data.cpp

```diff
@@ -417,4 +417,70 @@ void attributes(
     calc_attributes(horizon, surface, src_window, dst_window, attrs, from, to);
 }
 
+void align_surfaces(
+    RegularSurface const &primary,
+    RegularSurface const &secondary,
+    RegularSurface &aligned,
```

Comment:
How much validating should be left to the user here?

At the very least we should check that `aligned.size() == primary.size()`. But maybe also check the plane? Could either throw, or just update the aligned surface' plane to match primary